src/ol/Map.js

```diff
@@ -1136,7 +1137,12 @@ class Map extends BaseObject {
         // It's possible for the target to no longer be in the page if it has been removed in an
         // event listener, this might happen in a Control that recreates it's content based on
         // user interaction either manually or via a render in something like https://reactjs.org/
-        !(rootNode === doc ? doc.documentElement : rootNode).contains(target)
+        // skip check for maps place in ShadowRoot element structures
+        !(rootNode instanceof ShadowRoot)
+          ? !(rootNode === doc ? doc.documentElement : rootNode).contains(
+              target,
+            )
+          : false
```

Comment:
Why was this change necessary? I'm not sure if it's enough to just skip this check with a `ShadowRoot`. I think it would be better to use an equivalent logic in the `ShadowRoot` case, i.e. check if the shadowRoot's `documentElement.contains(target)`.

Reply:
It becomes necessary because `target` is the custom component element (`<ol-map>`) in the `ShadowRoot` case. Therefore `rootNode.contains(target)` would be false. IMHO it is necessary to check is part of the `documentElement`. I'd suggest for the following:

```js
      // check if event target is a child of the container that is no longer
      // in the page
      let targetInPage;
      if (rootNode instanceof ShadowRoot) {
        targetInPage = doc.documentElement.contains(target);
      } else {
        targetInPage = (
          rootNode === doc ? doc.documentElement : rootNode
        ).contains(target);
      }

      if (
        // Abort if the target is a child of the container for elements whose events are not meant
        // to be handled by map interactions.
        this.overlayContainerStopEvent_.contains(target) ||
        // Abort if the event target is a child of the container that is no longer in the page.
        // It's possible for the target to no longer be in the page if it has been removed in an
        // event listener, this might happen in a Control that recreates it's content based on
        // user interaction either manually or via a render in something like https://reactjs.org/
        // skip check for maps place in ShadowRoot element structures
        !targetInPage
      ) {
        return;
      }
```

What do you thinks @ahocevar ? Should I go for it or do you prefer an other check?

Reply:
I think something simpler should work too, maintaining the previous logic:
```js
const currentDoc = rootNode instanceof ShadowRoot
  ? rootNode.host
  : rootNode === doc ? doc.documentElement : rootNode)
if (!currentDoc.contains(target)) {
  return;
}

Reply:
Thanks for this @ahocevar, I picked this and added slightly extended in [03b95fa](https://github.com/openlayers/openlayers/pull/16094/commits/03b95fade0e00f96c877a0ca3285e68d22a81551)